### PR TITLE
Venue Maps

### PR DESCRIPTION
**Pull request by SimonShapiroMsft**
Venue Map Documentation PR

### CODE REVIEW

File: BingMaps/v8-web-control/TOC.yml [hunk 1]
```diff
@@ -696,6 +696,18 @@
               href: modules/traffic-module/trafficmanager-class.md
             - name: TrafficOptions Object
               href: modules/traffic-module/trafficoptions-object.md
+          name: Venue Maps Module
+          href: modules/venue-map-module/index.md
+          maintainContext: true 
+          items: 
+            - name: VenueMapFactory Class
```
Comment (Heptazane) [inline BingMaps/v8-web-control/TOC.yml:703]: Test comment

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+---------------------------------- | --------------------- | -----------------------------------
+`error` | function() | The callback function invoked after venue map footprints layer creation fails.
+`interactive` | boolean | When `true`, the footprints shown on this layer are interactive - meaning they respond to hover + tap events. This property is `true` by default.
+`success`| function(layer: **VenueMapFootprintsLayer** ) | The callback function invoked after a venue map footprints layer is succesfully created.
+`tileLoader` | **VenueMapTileLoader** | Method called to invoke loading of [venueTile](../../venues/venueTile.md) metadata. If not specified, `Network.downloadJsonp` is invoked.
+`tileUrl` | string OR function() => string | The custom url endpoint that returns a string or a function that returns a string. If the [venueTile](../../venues/venueTile.md) metadata is hosted on a different domain, and [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) is not enabled, but does support [JSONP](https://en.wikipedia.org/wiki/JSONP), the name of JSONP URL parameter that can be used to download the file across different domains needs to be specified.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md:30]: We need to revisit this:
1. I don't think there's any reason to have the function format of tileUrl. It doesn't take any parameters, so why wouldn't the caller just be able to put the string directly?
1. The documentation on CORS/JSONP was copied from GeoJSON.readFromUrl, which specifies the jsonp query param explicitly and performs same-origin requests without JSONP if the param is omitted. The current implementation for venue maps always uses JSONP. The documentation doesn't make sense here.
1. tileUrl should contain a {quadkey} placeholder that will be replaced by the LOD 10 quadkey to retrieve the footprint tiles in view.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ms.service: "bing-maps"
+---
+
+# VenueMapFootprintsLayerOptions
+Options used to customize how a [venueTile](../../venues/venueTile.md) file is read and loaded via the [VenueMapFactory](venuemapfactory-class.md).
+
+When hosting venue map footprints layer metadata at a custom endpoint, one needs to ensure that the data hosted at `tileUrl` can be loaded using the method defined in `tileLoader`. Specifically, one should pay attention to the case where their data is hosted on a different domain where CORS is not enabled. In this case, the hosting site should create another webpage triggered by an additional parameter (referred to below as the JSONP URL parameter) which allows for the file to be downloaded across domains.
+
+
+## Properties
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md:23]: All properties of VenueMapFootprintsLayerOptions are optional, though one of tileUrl | tileLoader must be present.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+`interactive` | boolean | When `true`, the footprints shown on this layer are interactive - meaning they respond to hover + tap events. This property is `true` by default.
+`success`| function(layer: **VenueMapFootprintsLayer** ) | The callback function invoked after a venue map footprints layer is succesfully created.
+`tileLoader` | **VenueMapTileLoader** | Method called to invoke loading of [venueTile](../../venues/venueTile.md) metadata. If not specified, `Network.downloadJsonp` is invoked.
+`tileUrl` | string OR function() => string | The custom url endpoint that returns a string or a function that returns a string. If the [venueTile](../../venues/venueTile.md) metadata is hosted on a different domain, and [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) is not enabled, but does support [JSONP](https://en.wikipedia.org/wiki/JSONP), the name of JSONP URL parameter that can be used to download the file across different domains needs to be specified. 
+
+
+<br/>
+<br/>
+
+## VenueMapTileLoader Method
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md:36]: VenueMapTileLoader is a method invoked by the footprints layer to request a tile for a region (identified by LOD 10 quadkey) in view. This is used to implement a custom data source for footprint tiles. The implementer should use the value of the quadkey param to identify a tile to load, then call the success function passing the corresponding tile data.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+Parameter                               | Req? | Description
+----------------------------------  | :---------------------: | -----------------------------------
+`error` | |The callback function invoked invoked when there's an error loading the data
+`quadKey` |✔ |Identifies the tile to load
+`success` | ✔ |The callback function invoked when the data for the venue tile is to be loaded
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapfootprintslayeroptions-object.md:41]: The parameters are out of order. Signature is (quadKey, success, error) => void

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+author: "simshap"
+ms.author: "simshap"
+manager: "cordellj"
+ms.service: "bing-maps"
+---
+
+# VenueMapOptions
+Options used to customize how a [Venue](../../venues/venue.md) file is read and loaded via the [VenueMapFactory](venuemapfactory-class.md).
+
+When hosting a venue map at a custom endpoint, one needs to ensure that the data hosted at `metadataUrl` can be loaded using the method defined in `metadataLoader`. Specifically, one should pay attention to the case where their data is hosted on a different domain where CORS is not enabled. In this case, the hosting site should create another webpage triggered by an additional parameter (referred to below as the JSONP URL parameter) which allows for the file to be downloaded across domains.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md:20]: Same issues as with tileUrl/tileLoader. It's one or the other; the two don't work together.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+## Properties
+Name                               | Type           | Description
+---------------------------------- | --------------------- | -----------------------------------
+`error` | function() | The callback function invoked after venue map creation fails.
+`metadataLoader` | **VenueMapMetadataLoader** | Method called to invoke loading of [Venue](../../venues/venue.md) metadata. If not specified, `Network.downloadJsonp` is invoked.
+`metadataUrl` | string OR function() => string | The custom url endpoint that returns a string or a function that returns a string. If the [Venue](../../venues/venue.md) file is hosted on a different domain, and [CORS](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing) is not enabled, but does support [JSONP](https://en.wikipedia.org/wiki/JSONP), the name of JSONP URL parameter that can be used to download the file across different domains needs to be specified. 
+`showFloorSwitcher` | boolean | When `true`, the floor switcher is deisplayed. This property is `false` by default.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md:28]: Typo: displayed. If 'true', the floor switcher control is shown when this venue is visible.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+`success`| function(VenueMap: [VenueMap](venuemap-class.md) ) | The callback function invoked after a venue map is succesfully created.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md:29]: Typo: successfully. We should fix the casing on the param, so it's function(venueMap: [VenueMap]). The function receives a reference to the created venue map.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemap-class.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+The **VenueMap** class represents a venue map in a specific instance of a [Map](../../map-control-api/map-class.md). The **VenueMap** provides functionality to toggle visibility and active floor. The **VenueMap** also exposes properties that can be leveraged for optimizing map view.
+
+
+## Methods
+
+Below are the list of methods for **VenueMap**.
+
+Name                               | Return Type           | Description
+---------------------------------- | --------------------- | -----------------------------------
+`dispose()` | | Disposes the venue map instance.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemap-class.md:27]: Disposes the venue map instance, removing it from the map.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapfactory-class.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+
+## Methods
+
+Below are the list of methods for **VenueMapFactory**.
+
+Name                               | Return Type           | Description
+---------------------------------- | ------------ | -----------------------------------
+`create(options:` [VenueMapOptions](venuemapoptions-object.md) `)` || Creates a [VenueMap](venuemap-class.md) with specifications defined in `options`.
+`createFootprintsLayer(options: ` [VenueMapFootprintsLayerOptions](venuemapfootprintslayeroptions-object.md) `)`| | Creates a [VenueMapFootprintsLayer](venuemapfootprintslayer-class.md) with specifications defined in `options`.
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapfactory-class.md:34]: I don't see documentation for VenueMapFootprintsLayer. Should include and document the venueclick event.

**Comment by rbrundritt** [comment]
#hold Please hold for internal discussion. This feature was purposely not made public in the past.

File: BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md [hunk 11]
```diff
@@ -28,18 +28,21 @@ Name                               | Type           | Description
...
-`showFloorSwitcher` | boolean | If `true`, the floor switcher control is shown when this venue is visible. This property is `false` by default.
-`success`| function(VenueMap: [VenueMap](venuemap-class.md) ) | The callback function invoked after a venue map is successfully created.
+`metadataUrl` | string | The custom url endpoint that returns a string or a function that returns a string. If the url contains the `{callback}` placeholder, it will be fetched as [JSONP](https://en.wikipedia.org/wiki/JSONP). Otherwise, it will be fetched as JSON using [XHR](https://en.wikipedia.org/wiki/XMLHttpRequest). 
+`showFloorSwitcher` | boolean | If `true`, the floor switcher control is shown when this venue is visible. This property is `false` by default. 
+`success`| function(venueMap: [VenueMap](venuemap-class.md) ) | The callback function invoked after a venue map is successfully created.
 `venueMapId` | string | The id of the venue map.
 
+> [!WARNING]
+> The floor switcher is only shown if the [navbar](https://docs.microsoft.com/en-us/bingmaps/v8-web-control/map-control-api/navigationbarmode-enumeration) is using the `default` style
```
Comment (drwestco) [inline BingMaps/v8-web-control/modules/venue-map-module/venuemapoptions-object.md:37]: 'square' style, not 'default'. It's weird we haven't documented square style. We have a public isdk sample for it: https://www.bing.com/api/maps/mapcontrol/isdk/squarenavigationbar?autoRedirect=false

**Comment by rbrundritt** [comment]
#needs-human-review

**Pull request merged**